### PR TITLE
[Merged by Bors] - feat(NhdsSet): lemmas from the Mandelbrot set connectedness project

### PR DESCRIPTION
**Pull request by urkud**
Add `eventually_nhdsSet_iff_exists`, `eventually_nhdsSet_iff_forall`,
and `IsOpen.mem_nhdsSet_self`.

Co-authored-by: @girving

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by girving** [comment]
Thank you!

File: Mathlib/Topology/NhdsSet.lean [hunk 1]
```diff
@@ -71,6 +71,17 @@ theorem mem_nhdsSet_iff_exists : s ∈ 𝓝ˢ t ↔ ∃ U : Set α, IsOpen U ∧
   rw [← subset_interior_iff_mem_nhdsSet, subset_interior_iff]
 #align mem_nhds_set_iff_exists mem_nhdsSet_iff_exists
 
+/-- A proposition is true on a set neighborhood of `s` if it is true on a larger open set -/
```
Comment (grunweg) [inline Mathlib/Topology/NhdsSet.lean:74]: Perhaps: `A proposition is true on a set neighborhood of `s` if and only if it is true on a larger open set.` ?

File: Mathlib/Topology/NhdsSet.lean [hunk 2]
```diff
@@ -71,6 +71,17 @@ theorem mem_nhdsSet_iff_exists : s ∈ 𝓝ˢ t ↔ ∃ U : Set α, IsOpen U ∧
...
+theorem eventually_nhdsSet_iff_exists {p : α → Prop} :
+    (∀ᶠ x in 𝓝ˢ s, p x) ↔ ∃ t, IsOpen t ∧ s ⊆ t ∧ ∀ x, x ∈ t → p x :=
+  mem_nhdsSet_iff_exists
+
+/-- A proposition is true on a set neighborhood of `s`
+if it is eventually true near each point in the set. -/
```
Comment (grunweg) [inline Mathlib/Topology/NhdsSet.lean:80]: Similarly, use `if and only if` here (or iff; I'm not sure if we use that in mathlib)?

**Review by grunweg — COMMENTED** [review]
Two drive-by comments, from a non-maintainer.

File: Mathlib/Topology/NhdsSet.lean (hunk 2 above)
Comment (ADedecker) [reply Mathlib/Topology/NhdsSet.lean:80]: `iff` is fine, but I agree.

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**